save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/topbar/TopBarUserField.kt

```diff
@@ -0,0 +1,141 @@
+@file:Suppress("FILE_NAME_MATCH_CLASS")
+
+package com.saveourtool.save.frontend.components.topbar
+
+import com.saveourtool.save.domain.Role
+import com.saveourtool.save.frontend.components.modal.logoutModal
+import com.saveourtool.save.frontend.externals.fontawesome.*
+import com.saveourtool.save.info.UserInfo
+import com.saveourtool.save.v1
+import com.saveourtool.save.validation.FrontendRoutes
+
+import csstype.ClassName
+import react.*
+import react.dom.aria.*
+import react.dom.html.ReactHTML.a
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.img
+import react.dom.html.ReactHTML.li
+import react.dom.html.ReactHTML.small
+import react.dom.html.ReactHTML.span
+import react.dom.html.ReactHTML.ul
+import react.router.useNavigate
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.isActive
+
+val topBarUserField = topBarUserField()
+
+/**
+ * [Props] of the top bar user field component
+ */
+external interface TopBarUserFieldProps : Props {
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
+}
+
+/**
+ * Displays the user's field
+ */
+@Suppress("MAGIC_NUMBER", "LongMethod", "TOO_LONG_FUNCTION")
+private fun topBarUserField() = FC<TopBarUserFieldProps> { props ->
+    val (isLogoutModalOpen, setIsLogoutModalOpen) = useState(false)
+    val (isAriaExpanded, setIsAriaExpanded) = useState(false)
```

Comment:
```suggestion
    val (isAriaExpanded, setAriaExpanded) = useState(false)
```

Reply:
Done [3484c44](https://github.com/saveourtool/save-cloud/pull/1540/commits/3484c445250a9b7f8c050fdb2a020d728d480a8e)

Reply:
I would actually prefer setter to be `set<STATE_NAME_IN_PASCAL_CASE>`

Reply:
@sanyavertolet, but `<STATE_NAME_IN_PASCAL_CASE>` is `[Aa]riaExpanded` in this case, isn't it?

Anyway, we have two patterns in our codebase:

 - `val (isFoo, setIsFoo) = useState(false)`, and
 - `val (isFoo, setFoo) = useState(false)`.
 
 The first one, IMO, is redundant, but I'll accept **any** decision.
 
 Let's just decide on the the unified coding style.

Reply:
@nulls

Reply:
let's migrate to
```kotlint
var isFoo by useState(false)
```

Reply:
Did it in [74eeb9a](https://github.com/saveourtool/save-cloud/pull/1540/commits/74eeb9a04e94a112770a0f49738ad7da6c97e63a)